### DISCUSSION TITLE
Need clarification on GitHub Actions

### DISCUSSION THREAD

**Discussion by behrangsa** [community/community#26997] [Actions]
<p>With GitHub Actions we can implement complete CI/CD workflows. But is there a way to display test, code coverage, and miscellaneous reports in a friendly HTML format like Jenkins and other CI/CD platforms?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8750915d8ee293c1808fd2c345403614b0aada91.png" data-download-href="/uploads/short-url/jj34BBpPSwRmgfoeupJ1k8jUQwh.png?dl=1" title="dMkkN.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/18451/181101103-a00c50b1-4e47-48f4-afd4-0b943d2322d8.png" alt="dMkkN.png" data-base62-sha1="jj34BBpPSwRmgfoeupJ1k8jUQwh" width="690" height="439" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8750915d8ee293c1808fd2c345403614b0aada91_2_690x439.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8750915d8ee293c1808fd2c345403614b0aada91_2_1035x658.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8750915d8ee293c1808fd2c345403614b0aada91.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8750915d8ee293c1808fd2c345403614b0aada91_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">dMkkN.png</span><span class="informations">1094×697 128 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Or is it recommended to offload testing to another platform (e.g. Jenkins, Circle CI, etc.)?</p>
<p>What about artifact management?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6bb3fa2db92af3ae00e94c0ce31d86a72186493b.png" data-download-href="/uploads/short-url/fmMGHTLJ7W7uthVJ8oro09aPJeX.png?dl=1" title="art.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/18451/181101104-1b019bdb-de6b-4ee2-804e-50fbac0a1d45.png" alt="art.png" data-base62-sha1="fmMGHTLJ7W7uthVJ8oro09aPJeX" width="690" height="232" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6bb3fa2db92af3ae00e94c0ce31d86a72186493b_2_690x232.png, https://aws1.discourse-cdn.com/github/original/2X/6/6bb3fa2db92af3ae00e94c0ce31d86a72186493b.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/6bb3fa2db92af3ae00e94c0ce31d86a72186493b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6bb3fa2db92af3ae00e94c0ce31d86a72186493b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">art.png</span><span class="informations">960×324 66.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by lee-dohm** [comment, marked as answer]
<p>Good question!</p>
<p>GitHub Actions is intended to be very open-ended to allow you to implement whatever kind of workflow you can imagine. In order to be this open-ended though, that means that whoever implements the Actions in the workflow will have to build the code, components, or whatever other infrastructure is necessary to display the reports that you’re describing. So the workflow can execute the task of running tests, executing code coverage, collecting their reports, and uploading them to a host somewhere they can be referred back to later. But that hosting will have to be somewhere outside of the workflow since a workflow is time-limited. The same is true for build artifact management.</p>
<p>I hope that helps!</p>

**Comment by webknjaz** [comment]
<p>FTR you can achieve some of that by posting Markdown via Checks API. I guess you could try embedding images as base64-encoded data-urls…</p>

**Comment by mittalyashu** [comment]
<p>I don’t see GitHub actions on my GitHub repositories.</p>
<p><a href="https://github.com/gitthermal/thermal" rel="nofollow noopener">https://github.com/gitthermal/thermal</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3a35e802aaefd5901f586df5174d9b8564fcd7e7.png" data-download-href="/uploads/short-url/8iXbAKRvwmVdQJ6Fd1pvM4Eilnh.png?dl=1" title="msedge_Y6fPmrCvNE.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29014463/181101109-37ddb47a-284f-4107-a66a-5249b1040d9b.png" alt="msedge_Y6fPmrCvNE.png" data-base62-sha1="8iXbAKRvwmVdQJ6Fd1pvM4Eilnh" width="690" height="71" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3a35e802aaefd5901f586df5174d9b8564fcd7e7_2_690x71.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/3a35e802aaefd5901f586df5174d9b8564fcd7e7_2_1035x106.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/3a35e802aaefd5901f586df5174d9b8564fcd7e7_2_1380x142.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3a35e802aaefd5901f586df5174d9b8564fcd7e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">msedge_Y6fPmrCvNE.png</span><span class="informations">1529×158 16.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by lee-dohm** [comment]
<p><a class="mention" href="/mittalyashu">@mittalyashu</a> You have to <a href="https://github.com/features/actions" rel="nofollow noopener">sign up for beta access</a> first.</p>

**Comment by ssbarnea** [comment]
<p>I guess that is a posh way of confirming that there is no support for hosting any build artifacts. The fact that there is an action to create a zip file does not bring much value, we were already able to do the same using curl or whatever tool we wanted.</p>
<p>Still, ability to browse build artifacts, like html reports is a very basic functionality for any CI/CD tool. A zip file is anything but user friendly, as it would require user to download the file, unzip it in order to look at it. Should I also mention that sharing links to a report now becomes sharing links to a zip file… hmm. I doubt many people would find “zip” approach as being practical, unless you really produce binaries.</p>

**Comment by saltcreek-alex** [comment]
<p>I agree with <a class="mention" href="/ssbarnea">@ssbarnea</a></p>
<p>In some of our private repositories, we generate an API documentation as part of the CI process.<br>
Currently, we upload the HTML for the docs along with the test reports as a ZIP, which has to be downloaded, everytime someone wants to look up the latest docs. </p>
<p>Other CI/CD tools provide this functionality and I think that this would make the GitHub CI/CD workflow a lot more usable.</p>
